inventory/by_environment/production

```diff
@@ -0,0 +1,85 @@
+[production]
```

Comment:
Could we use [groups of groups](https://docs.ansible.com/ansible/2.7/user_guide/intro_inventory.html#groups-of-groups-and-group-variables) here, rather than listing all the servers in multiple files?

The goal would be to reduce the burden of maintaining these, so you could add a new VM to an existing group, say libwww_prod, and it would automatically be included in the production and apache_servers groups.

For example:

```
[production:children]
abid_production
annotations_production
approvals_production
bibdata_production
...
```

Reply:
That's a great idea. We'd have to be sure every machine is in some group somewhere, but I don't think that's too far of a reach.

I did enjoy alphabetizing and adding comments, though.

Reply:
I've noted this in the ADR, but I'll put it here on the PR as well:

Ansible loads inventory in alphabetical order (well, linux/ASCII order). You cannot create a parent group until after all the child groups are loaded. For this reason, the per-project files need to be listed before the per-environment files. I've created a sub-directory for the per-project inventory files called `all_projects`. 

If you change it to `projects` or `by_project`, the child groups will be loaded later, the `[production]` and `[qa]` and `[staging]` groups will not work, and you will see the error message `Unable to parse /path/to/inventory/by_environment/ as an inventory source`.